src/http_server/http_server.rs

```diff
@@ -1,37 +1,67 @@
 use axum::{routing::get, Router};
-
+use log::info;
 use crate::http_server::routes;
 
+#[derive(Clone, Default)]
 pub struct HttpServer {
     host: String,
     port: String,
+    router: Router,
 }
 
-impl HttpServer {
-    pub fn new(host: String, port: String) -> HttpServer {
-        HttpServer { host, port }
-    }
-    pub async fn server_start(self) {
-        let (app, addr) = HttpServer::init_server_configuration(self).await;
+#[derive(Clone, Default)]
+pub struct HttpServerBuilder {
+    host: Option<String>,
+    port: Option<String>,
+    router: Router,
+}
 
-        axum::Server::bind(&addr.parse().unwrap())
-            .serve(app.into_make_service())
-            .await
-            .unwrap();
+impl HttpServerBuilder {
+    pub fn new() -> Self {
+        HttpServerBuilder::default()
     }
 
-    async fn init_server_configuration(self) -> (Router, String) {
-        let app = HttpServer::init_basic_routes();
-        let addr = format!("{}:{}", self.host, self.port);
+    pub fn host(mut self, host: impl Into<String>) -> Self {
+        self.host = Some(host.into());
+        self
+    }
 
-        return (app.clone(), addr);
+    pub fn port(mut self, port: impl Into<String>) -> Self {
+        self.port = Some(port.into());
+        self
```

Comment:
It would be preferable to directly inject the configuration via the `ConfigurationWrapper` then use it inside the `build` method
```suggestion
pub fn configuration_wrapper(mut self, configuration_wrapper: impl Into<ConfigurationWrapper>) {
    self.host = Some(configuration_wrapper.into());
}
```
or something among those lines

Reply:
Yeah but no, I would like to be independent in my code like if you change yours, I don't want to change mine, I'm sure I understand

Reply:
No sorry I don't get it, it's your responsibility to make your code configurable by the standard way (just like my_files or the  file_watcher implementation cc @majent4 ).
Configuration management should be done by the callee and not the caller, otherwise you're just giving the trouble of configurating your code to some else

Reply:
On a side note, I don't want to change my code either if yours change (see the vicious circle?)

Reply:
Any suggest @majent4  ?

Reply:
> Any suggest @majent4 ?

@Cavonstavant implemented the ```ConfigurationWrapper``` for this purpose. I won't approve this PR if you don't use it.

Reply:
> > Any suggest @majent4 ?
> 
> @Cavonstavant implemented the `ConfigurationWrapper` for this purpose. I won't approve this PR if you don't use it.

that was'nt the question dude